Source/Core/Core/HW/WiimoteEmu/WiimoteEmu.h

```diff
@@ -354,4 +411,77 @@ class Wiimote : public ControllerEmu::EmulatedController, public WiimoteCommon::
 
   Config::ConfigChangedCallbackID m_config_changed_callback_id;
 };
+
+class BalanceBoard final : public WiimoteBase
+{
+  friend class BalanceBoardExt;
+
+public:
+  static constexpr u16 BUTTON_POWER = Wiimote::BUTTON_A;
+
+  explicit BalanceBoard(u8 index);
+
+  ControllerEmu::ControlGroup* GetBalanceBoardGroup(BalanceBoardGroup group) const;
+
+  std::string GetName() const override { return "BalanceBoard"; }
+  void LoadDefaults(const ControllerInterface& ciface) override;
+
+  void PrepareInput(WiimoteEmu::DesiredWiimoteState* target_state,
+                    SensorBarState sensor_bar_state) override;
+  void Update(const WiimoteEmu::DesiredWiimoteState& target_state) override;
+  WiimoteCommon::ButtonData GetCurrentlyPressedButtons() override;
+
+  void Reset() override;
+  void LoadDefaultEeprom() override {}
+
+  void DoState(PointerWrap& p) override;
+
+  Extension* GetActiveExtension() const override
+  {
+    // XXX const_cast seems wrong here, but how does Wiimote's version work then?
+    return const_cast<Extension*>(static_cast<const Extension*>(&m_ext));
+  }
+
+  ExtensionNumber GetActiveExtensionNumber() const override
+  {
+    return ExtensionNumber::BALANCE_BOARD;
+  }
+
+  virtual ControllerEmu::SubscribableSettingValue<bool>& GetMotionPlusSetting() override
+  {
+    return m_fake_motion_plus_setting;
+  }
+
+  virtual ControllerEmu::SubscribableSettingValue<int>& GetAttachmentSetting() override
+  {
+    return m_fake_attachment_setting;
+  }
```

Comment:
This is bad design, but was the easiest way to make the tas input window work. It probably should be changed later.